### ISSUE TITLE
LSP should maybe ignore .json files by default

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno#21409]
[User reports](https://github.com/denoland/vscode_deno/issues/895#issuecomment-1832712608) that using a 8.4Mb JSON file causes a slow down in the LSP. We should investigate how this file in used in the project (is it imported somewhere?) and figure out how to speed it up. Maybe we should disable analysis of JSON files altogether?

**Comment by marvinhagemeister** [comment]
Wondering if it would make sense to parse that file in rust, generate a simple definition file in rust and then hand it over to TSC. Looking at the JSON file the type would be:

```ts
export default Array<[number, {
  key: number,
  value: {
    id: number,
    first: string,
    last: string,
    age: number
  }
}]>
```

**Comment by nhrones** [comment]
The app repo is https://github.com/nhrones/BuenoCache
The json file is /workers/hundredk.json
It's imported in /workers/buildWorker.js

Note:  This is not a Deno project!
Deno is enabled in this project to support my custom Deno-dev-server -- (HotServe).](https://github.com/nhrones/Hot)

**Comment by nhrones** [comment]
Heads up:
Even with the following settings disabling this file, I still have issues.
```json
{
   "deno.enable": true,
   "deno.lint": true,
   "deno.unstable": true,
   "deno.disablePaths": ["./hundredk.json", "./.vscode"],
}
```
If I actually open the file in VS Code, I get the OOM crash.  
```
#
# Fatal JavaScript out of memory: Reached heap limit
#
```
Note: I rarely open this file,  It was auto generated to be used for DB testing.

**Comment by nayeemrmn** [comment]
JSON files already aren't preloaded by default unless they are imported, in which case their types need to be analysed. We determined this wasn't the cause for most users, just the isolated case linked here. Closing.

**Comment by nayeemrmn** [comment]
> Still getting OOM crash, even without any large files.

So that's not related to the JSON file, right? I think this particular issue isn't worth pursuing.

**Comment by nhrones** [comment]
> > Still getting OOM crash, even without any large files.
> 
> So that's not related to the JSON file, right? I think this particular issue isn't worth pursuing.

Yes I agree.  In my apps with large json files, I've been able to isolate this issue with your recommended fix
```ts
{
   "deno.enable": true,
   "deno.disablePaths": ["./dist","./.vscode", "./data"],
}
``` 
I only wanted to provide feedback that on my laptop, I'm still getting random LSP crash.  
The logs report that the LSP has used 3.2GB (all available memory), and resets.  After the third reset (hours of editing in same project), The report is LSP-OOM - LSP will not be reset!  
At that point I just restart VS Code.

**Comment by nayeemrmn** [comment]
The above report wasn't related to JSON files
